pkg/promtail/promtail_test.go

```diff
@@ -350,17 +350,23 @@ func buildTestConfig(t *testing.T, positionsFileName string, logDirName string)
 		t.Fatal("Failed to parse client URL")
 	}
 
-	clientConfig := client.Config{
-		URL:            clientURL,
-		BatchWait:      10 * time.Millisecond,
-		BatchSize:      10 * 1024,
-		ExternalLabels: nil,
+	cfg := config.Config{
+		ServerConfig:    server.Config{},
+		ClientConfig:    client.Config{},
+		PositionsConfig: positions.Config{},
+		ScrapeConfig:    []scrape.Config{},
+		TargetConfig:    targets.Config{},
 	}
```

Comment:
This is all a noop - they are all zero values.

Reply:
Strange, I thought I needed those to call `flagext.RegisterFlags(&cfg)` but it appears I do not, I will remove them.